Author:     Andy Shaw <andy.shaw@theqtcompany.com>
AuthorDate: 2015-06-20 01:14:07 +0200
Commit:     Andy Shaw <andy.shaw@theqtcompany.com>
CommitDate: 2015-10-12 23:16:07 +0200

Include the qml files into a resource when building a static plugin

Whenever a static qml plugin is made then the qmldir and other related
files needs to be compiled into the resources so that it can be found
in the import path. By using qrc:/qt-project.org/imports we can have
this taken care automatically for us via the build system, leaving us
to just ensure that it is initialized in the code.

Task-number: QTBUG-35754
Task-number: QTBUG-39462
Change-Id: Ifa7e2a66fd78dc6713dd7a8661ea2c155b174d35

--- a/qtbase/mkspecs/features/qml_module.prf.orig	2015-11-02 14:30:16.074161000 +0300
+++ b/qtbase/mkspecs/features/qml_module.prf	2015-11-02 14:31:03.622405700 +0300
@@ -42,6 +42,17 @@
     QMAKE_EXTRA_COMPILERS += qmlfiles2build
 }
 
+# Ensure the qmldir is included in the resources
+!qml1_target:static {
+    URITARGET = $$replace(URI, "\.", "_")
+    !isEmpty(fq_qml_files) {
+        $${URITARGET}.files += $$fq_qml_files
+        # qrcImportPath is the path used for locating imports inside the resources
+        $${URITARGET}.prefix = /qrcImportPath/$$TARGETPATH
+        RESOURCES += $$URITARGET
+    }
+}
+
 # Install rules
 qmldir.base = $$_PRO_FILE_PWD_
 qmldir.files = $$fq_qml_files
